### PR TITLE
📄handle YAML special string value

### PR DESCRIPTION
**Pull request by mingwli**
#4104

### CODE REVIEW

File: src/docfx/lib/YamlUtility.cs [hunk 1]
```diff
@@ -144,23 +145,21 @@ private static JToken ToJson(YamlNode node)
...
                     {
                         return PopulateLineInfoToJToken(new JValue(n), node);
                     }
-                    if (double.TryParse(scalar.Value, out var d))
+                    if (double.TryParse(scalar.Value, NumberStyles.Any, CultureInfo.InvariantCulture, out var d))
                     {
                         return PopulateLineInfoToJToken(new JValue(d), node);
                     }
-                    if (bool.TryParse(scalar.Value, out var b))
+                    if (bool.TryParse(scalar.Value,out var b))
```
Comment (yufeih) [inline src/docfx/lib/YamlUtility.cs]: <space>

**Comment by mingwli** [comment]
The explicit null is resolved as null value in v2.
https://raw.githubusercontent.com/openpublish/docfx-migration-to-v3/null-value/docfx-v3-migration/schema-syntax-check/others/null-value/explicit/a.md
image

File: test/docfx.Test/lib/YamlUtilityTest.cs [hunk 2]
```diff
@@ -158,6 +158,37 @@ public void TestBoolean()
             Assert.Equal(new[] { true, true, true, false, false, false }, value2.Select(j => (bool)j).ToArray());
         }
 
+        [Fact]
+        public void TestNull()
```
Comment (live1206) [inline test/docfx.Test/lib/YamlUtilityTest.cs]: better to use `InlineData` to specify input and output? would be easier to add more test cases

Comment (mingwli) [reply test/docfx.Test/lib/YamlUtilityTest.cs]: updated

**Pull request merged**